pkg/scheduler/controller/shoot/reconciler.go

```diff
@@ -141,6 +144,10 @@ func determineSeed(
 	if err != nil {
 		return nil, err
 	}
+	filteredSeeds, err = filterMultiZonalSeeds(filteredSeeds, shoot)
```

Comment:
It would be good to have another test case for this in https://github.com/gardener/gardener/tree/master/test/integration/scheduler.

Reply:
Sure, I will add a test for the same.